src/auth/channel.ts

```diff
@@ -19,7 +19,7 @@ export type ChannelData = string | ArrayBufferLike | Blob | ArrayBufferView
 export const createWssChannel = async (options: ChannelOptions): Promise<Channel> => {
   const { username, handleMessage } = options
 
-  const rootDid = await await did.root(username).catch(() => null)
+  const rootDid = await waitForRootDid(username)
```

Comment:
Should this not run `did.root()` immediately, instead of waiting 2 seconds the first time around?

Reply:
You are right. Creating a new PR to fix that.